compiler/rustc_codegen_llvm/src/debuginfo/metadata.rs

```diff
@@ -437,19 +438,23 @@ fn pointer_or_reference_metadata<'ll, 'tcx>(
     unique_type_id: UniqueTypeId,
 ) -> MetadataCreationResult<'ll> {
     let pointee_type_metadata = type_metadata(cx, pointee_type);
+    let pointee_tail_type = cx.tcx.struct_tail_erasing_lifetimes(pointee_type, cx.param_env());
```

Comment:
Can you move this into `fat_pointer_kind()`? Then users of that function don't have to know about `struct_tail_erasing_lifetimes()`.